adwords_client/internal_api/builder.py

```diff
@@ -54,45 +56,152 @@ def cast_operation(self, operation):
+            elif object_type == 'billing_account' and sync:
+                yield {}
             else:
                 logger.warning('Operation not recognized: {}', operation)
                 yield None
 
-    def _parse_keyword(self, operation):
-        yield keyword.new_keyword_operation(**operation)
+    def _parse_campaign_criterion_operation(self, operation, is_get_operation):
+        if is_get_operation:
+            yield campaign_criterion.get_campaign_criterion_operation(**operation)
+        elif operation['object_type'] == 'campaign_ad_schedule':
+            yield campaign_criterion.ad_schedule_operation(**operation)
+        else:
+            raise NotImplementedError()
 
-    def _parse_adgroup(self, operation):
-        yield adgroup.adgroup_operation(**operation)
+    def _parse_callouts_setting_for_campaign(self, operation, is_get_operation):
+        if is_get_operation:
+            yield campaign_extensions_setting.get_campaign_extension_operation(**operation)
+        else:
+            yield campaign_extensions_setting.callout_setting_for_campaign_operation(**operation)
 
-    def _parse_ad(self, operation):
-        yield ad.expanded_ad_operation(**operation)
+    def _parse_structured_snippets_setting_for_campaign(self, operation, is_get_operation):
+        if is_get_operation:
+            yield campaign_extensions_setting.get_campaign_extension_operation(**operation)
+        else:
+            yield campaign_extensions_setting.structured_snippet_setting_for_campaign_operation(**operation)
 
-    def _parse_campaign(self, operation):
-        if operation.get('operator', 'ADD').upper() == 'ADD' and 'budget_id' not in operation:
-            operation['budget_id'] = cast_to_adwords('budget_id', self.get_next_id())
-            yield campaign.add_budget(**operation)
-        yield campaign.campaign_operation(**operation)
-        for language_id in operation.get('languages', []):
-            language_id = cast_to_adwords('language_id', language_id)
-            yield campaign.add_campaign_language(language_id=language_id, **operation)
-        for location_id in operation.get('locations', []):
-            location_id = cast_to_adwords('location_id', location_id)
-            yield campaign.add_campaign_location(location_id=location_id, **operation)
+    def _parse_sitelinks_setting_for_campaign(self, operation, is_get_operation):
+        if is_get_operation:
+            yield campaign_extensions_setting.get_campaign_extension_operation(**operation)
+        else:
+            yield campaign_extensions_setting.sitelink_setting_for_campaign_operation(**operation)
 
-    def _parse_label(self, operation):
-        yield label.new_label_operation(**operation)
+    def _parse_attach_label(self, operation, is_get_operation):
+        if is_get_operation:
+            raise Exception('There is not get method for this object type')
```

Comment:
When in doubt, raise a [`RuntimeError`](https://docs.python.org/3.6/library/exceptions.html#RuntimeError), not an [`Exception`](https://docs.python.org/3.6/library/exceptions.html#Exception).

Better to raise a [TypeError](https://docs.python.org/3.6/library/exceptions.html#TypeError)  or a [ValueError](https://docs.python.org/3.6/library/exceptions.html#ValueError) in this specific case. Check the links for explanation of what they mean.

Best would be raising a custom subclass of `TypeError` or `ValueError`, but no need to go this far for now.